### PR TITLE
Smoothing and annotations from periods

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds light curve smoothing (analogous to what is available for spectroscopy) and annotation posting from periods.

### CODE REVIEW

File: static/js/plotjs/foldphase.js [hunk 1]
```diff
@@ -18,6 +50,76 @@ for (let i = 0; i < n_labels; i++) {
...
+    for (j = 0; j < alist.length; j++) {
+      amag_sort[j] = alist[j].mag;
+    }
+    const amag_sort_smooth = smoothing_func(amag_sort, binsize);
+    const amag_sort_smooth_reordered = [];
+    for (var j = 0; j < folda.data.mjd_folda.length; j++) {
+      for (var k = 0; k < folda.data.mjd_folda.length; k++) {
+        if (j == alist[k].index) {
+          amag_sort_smooth_reordered.push(amag_sort_smooth[k]);
+          foldaerr.data.ys[j] = [amag_sort_smooth[k], amag_sort_smooth[k]];
```
Comment (guynir42) [inline static/js/plotjs/foldphase.js:83]: What is happening here?

File: static/js/plotjs/foldphase.js [hunk 2]
```diff
@@ -18,6 +50,76 @@ for (let i = 0; i < n_labels; i++) {
...
+        : 1;
+    });
+    const amag_sort = [];
+    for (j = 0; j < alist.length; j++) {
+      amag_sort[j] = alist[j].mag;
+    }
+    const amag_sort_smooth = smoothing_func(amag_sort, binsize);
+    const amag_sort_smooth_reordered = [];
+    for (var j = 0; j < folda.data.mjd_folda.length; j++) {
+      for (var k = 0; k < folda.data.mjd_folda.length; k++) {
```
Comment (guynir42) [inline static/js/plotjs/foldphase.js:80]: There's no way to sort a list of tuples in JS? this double loop can't be good for you...

File: static/js/plotjs/foldphase.js [hunk 3]
```diff
@@ -18,6 +50,76 @@ for (let i = 0; i < n_labels; i++) {
     foldb.data.mjd_foldb[m] = folda.data.mjd_folda[m] + 1;
     foldberr.data.xs[m] = [foldb.data.mjd_foldb[m], foldb.data.mjd_foldb[m]];
   }
+
+  if (binsize > 1) {
+    var j;
```
Comment (guynir42) [inline static/js/plotjs/foldphase.js:55]: What if you push the whole sorting mechanism into the smoothing function and then you'll just call it for 'a' and 'b' separately?

File: static/js/plotjs/foldphase.js (hunk 1 above)
Comment (mcoughlin) [reply static/js/plotjs/foldphase.js:83]: The y-array is just the same value repeated twice... for some reason. Figured it out in console.log.

File: static/js/plotjs/foldphase.js (hunk 2 above)
Comment (mcoughlin) [reply static/js/plotjs/foldphase.js:80]: You ask me questions like I know JS ;)

File: static/js/plotjs/foldphase.js (hunk 3 above)
Comment (mcoughlin) [reply static/js/plotjs/foldphase.js:55]: I think the main issue here is in the variable names

Comment (guynir42) [reply static/js/plotjs/foldphase.js:55]: What do you mean? you are doing the exact same logic (sorting then smoothing) on two arrays. 
Why not put each array into a function that returns the sorted+smoothed values?

Comment (mcoughlin) [reply static/js/plotjs/foldphase.js:81]: Ah got ya. Ok implemented.

File: static/js/plotjs/foldphase.js [hunk 4]
```diff
@@ -22,6 +22,32 @@ const smoothing_func = (values, window_size) => {
...
+  for (l = 0; l < list.length; l += 1) {
+    for (k = 0; k < list.length; k += 1) {
+      if (l === list[k].index) {
+        mag_sort_smooth_reordered.push(mag_sort_smooth[k]);
+        break;
+      }
+    }
+  }
+
+  return mag_sort_smooth_reordered;
```
[the remark this answers is not in the document]
Comment (guynir42) [reply static/js/plotjs/foldphase.js:48]: Please make sure to test this is correct, my JS is not the best either...

**Pull request merged**